docs/kbs_protocol.md

```diff
@@ -0,0 +1,265 @@
+
+# Introduction
+
+The purpose of the attestation between KBS and KBC is actually to confirm whether 
+**the platform where the KBC is located is in the expected security state**, that is, it runs in 
+a real and harmless HW-TEE.
+
+In Universal-KBS, HW-TEE attestation process is the semantics of the application 
+layer, which is defined as a simple, universal and extensible 
+"Request-Challenge-Attestation-Response" (RACR) protocol. The temporary asymmetric key generated 
+by HW-TEE is used to encrypt the response payload, and the token mechanism is used to avoid the 
+performance problems caused by multiple attestation.
+
+In order to ensure the ease of use and security completeness of Universal-KBS, we will use HTTPS 
+as the default transmission method to carry the application layer semantics designed in this 
+document. This is because HTTPS provides KBC with a means to authenticate KBS identity, which 
+effectively avoids malicious attackers from hijacking KBS address to impersonate KBS and deceive KBC.
+In order to achieve this, the public key of KBS needs to be transmitted to KBC through an effective way, 
+and the specific way of public key distribution is not within the scope of this document.
+In addition, it should be noted that the confidentiality protection provided by HTTPS alone is not enough 
+to meet the security requirements of KBS protocol, therefore, as mentioned above, the key generated 
+by hw-tee needs to be used to encrypt and protect the confidential data.
+
+# RCAR semantics
+
+The semantics of attestation defined by Universal-KBS is a simple and extensible four-step model, 
+which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC actively sends to KBS, calls the service API provided by KBS to request resources or services.
+2. **Challenge**: After receiving the request, KBS returns a challenge to KBC and asks KBC to 
+send evidence to prove that its environment (HW-TEE) is safe and reliable.
+3. **Attestation**: KBC sends the evidence and the newly generated public key in HW-TEE to KBS.
+4. **Response**: After the KBS verified the evidence, it returns the API output of the requested 
+service and a token to KBC, where the API output payload is protected by the public key in step 3. 
+Within the valid time of the token, KBC can directly request resources or services of KBS by 
+virtue of the token without step 2 and step 3.
+
+## Request
+
+The payload format of the request is as follows:
+
+(Note that the `/*...*/` comments are not valid in JSON, and must not be used in real message.)
+
+```json
+{
+    "request": {
+	    /* Protocol version number used by KBC */
+	    "protocol-version": "0.1.0",
+	    /* Types of HW-TEE platforms where KBC is located, such as "TDX", "SEV-SNP", etc. */
+	    "tee": "",
+	    /* Access token to avoid multiple attempts triggered by consecutive requests. It can be set to blank. */
```

Comment:
I think the semantic of the token should be clarified. Saying that it *can* be set to blank is a bit confusing. When should KBC set it to blank and when shouldnt it?

IIUC there are only 2 supported cases:

1. A token is provided, which means a previous attestation was done. If the token is still valid, the challenge/attestation part can be skipped. If it is not valid, KBS will send a Challenge
2. Token is left blank, which means the KBC is implictly expecting to go through attestation and KBS will reply with a Challenge.